### PR TITLE
PHPCodeSnifferBear: Disable two kinds of error

### PR DESCRIPTION
**Pull request by bhavishyagopesh**
Adds and sets `absolute_line_length` to '0'.So as to disable two kinds
of errors.

Closes https://github.com/coala/coala-bears/issues/1950

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by bhavishyagopesh** [comment]
@sils / @Makman2 does this PR needs any more changes?

**Comment by Makman2** [comment]
@bhavishyagopesh please don't ping maintainers personally for reviews (or any other person specifically). Anybody mentioned somewhere will get subscribed to the thread and the messages are appearing in the "Participating" list.

**Comment by alphadose** [comment]
ack 35617562e5e9a4948bac3ae9850c572f8a0618ec

**Comment by Makman2** [comment]
Tests tests tests! :D

**Comment by Makman2** [comment]
Imo the commit message is a bit confusing: `Disable two kinds of error`
This sounds like you are just disabling two error types (which sound to be different)
I think you should point out that you managed to remove a redundancy in the checks, like `Deduplicate results` or similar

**Comment by Makman2** [comment]
\+ your commit body could be more descriptive and less redundant ;)
(Also add a space after period! :D)

**Comment by Makman2** [comment]
unack 3561756

**Comment by bhavishyagopesh** [comment]
@Makman2  I'm doubtful about tests, I'm just adding a constant in config so how exactly should I test this?

**Comment by Makman2** [comment]
You have to construct a case where the type of error described in the issue occurs twice. After your change it should be only displayed once, right? If we don't test it we can't know whether the bug sneaks in again.

**Comment by bhavishyagopesh** [comment]
/process/pending

File: tests/php/PHPCodeSnifferBearTest.py [hunk 1]
```diff
@@ -39,8 +39,34 @@
 """
 
 
+bad_file_with_more_max_line_length = """<?php
+/**
+ * PHP_CodeSniffer tokenizes PHP code and detects violations of a defined \
+set of coding.This is too to test whether the absoluteLineLength.
```
Comment (Makman2) [inline tests/php/PHPCodeSnifferBearTest.py]: not important, but the sentence kinda ends abruptly :P (just noting :P)

Comment (bhavishyagopesh) [reply tests/php/PHPCodeSnifferBearTest.py]: fixed

**Comment by Makman2** [comment]
I tested your stuff locally, once with all your changes and once removing only the code inside the bear (but keeping your test). Both times the tests passes, so it seems your new test doesn't detect the issue you address.

**Comment by bhavishyagopesh** [comment]
@Makman2  I changed the test-procedure, you might wanna check now...

File: tests/php/PHPCodeSnifferBearTest.py [hunk 2]
```diff
@@ -44,3 +62,19 @@
...
     tempfile_kwargs={'suffix': '.php'})
+
+
+class PHPCodeSnifferBearLineLengthTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section('name')
+        self.uut = PHPCodeSnifferBear(self.section, Queue())
+        test_files = os.path.join(os.path.dirname(
+            __file__), 'codesniffer_test_files')
```
Comment (Makman2) [inline tests/php/PHPCodeSnifferBearTest.py]: you should use `get_testfile_path` ;)

File: tests/php/PHPCodeSnifferBearTest.py [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+def get_testfile_path(name):
+    return os.path.join(os.path.dirname(__file__),
+                        'codesniffer_test_files',
+                        name)
+
+
+def load_testfile(name):
+    with open(get_testfile_path(name)) as f:
+        output = f.readlines()
+    return output
```
Comment (Makman2) [inline tests/php/PHPCodeSnifferBearTest.py]: you can also do `return f.readlines()`

Comment (Makman2) [reply tests/php/PHPCodeSnifferBearTest.py]: (inside the `with` stmt)

File: tests/php/PHPCodeSnifferBearTest.py [hunk 4]
```diff
@@ -44,3 +62,19 @@
...
+        self.uut = PHPCodeSnifferBear(self.section, Queue())
+        test_files = os.path.join(os.path.dirname(
+            __file__), 'codesniffer_test_files')
+        self.config_file = os.path.join(test_files, 'config.xml')
+        self.long_length_file = os.path.join(
+            test_files, 'long_line_length.php')
+
+    def test_absolute_line_length(self):
+        self.section.append(Setting('max_line_length', '160'))
+        self.check_validity(self.uut, [], self.long_length_file)
```
Comment (Makman2) [inline tests/php/PHPCodeSnifferBearTest.py]: I think you can probably use `verify_local_bear` in this case, makes your test more compact :+1:

**Comment by bhavishyagopesh** [comment]
@Makman2  the old version is passing `verify_local_bear` ..

File: tests/php/PHPCodeSnifferBearTest.py [hunk 5]
```diff
@@ -44,3 +56,16 @@
...
     invalid_files=(bad_file,),
     tempfile_kwargs={'suffix': '.php'})
+
+
+class PHPCodeSnifferBearLineLengthTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section('name')
+        self.uut = PHPCodeSnifferBear(self.section, Queue())
+        self.config_file = get_testfile_path('config.xml')
```
Comment (Makman2) [inline tests/php/PHPCodeSnifferBearTest.py]: unused variable

File: tests/php/PHPCodeSnifferBearTest.py [hunk 6]
```diff
@@ -44,3 +56,16 @@
...
+        self.long_length_file = get_testfile_path('long_line_length.php')
```
Comment (Makman2) [inline tests/php/PHPCodeSnifferBearTest.py]: here you just get the path string, not the file content^^

**Comment by Makman2** [comment]
> @Makman2 the old version is passing verify_local_bear ..

This doesn't work?

```python
PHPCodeSnifferBearLineLengthTest = verify_local_bear(
    PHPCodeSnifferBear,
    valid_files=(long_length_test_file,),  # <-- note to put the contents of the test file, not the path
    settings={'max_line_length', '160'}
    tempfile_kwargs={'suffix': '.php'})
```

**Comment by bhavishyagopesh** [comment]
@Makman2  made the required changes

**Comment by Makman2** [comment]
ack e98488e

**Comment by Makman2** [comment]
ack e98488e

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Pull request merged**